drivers/auxiliary/Terrans_PowerBoxGo_V2.h

```diff
@@ -0,0 +1,125 @@
+  the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+  Boston, MA 02110-1301, USA.
+
+  The full GNU General Public License is included in this distribution in the
+  file called LICENSE.
+*******************************************************************************/
+
+#pragma once
+
+#include "defaultdevice.h"
+
+#include <vector>
+#include <stdint.h>
+
+#include "basedevice.h"
+
+namespace Connection
+{
+  class Serial;
+}
+
+class TerransPowerBoxGoV2 : public INDI::DefaultDevice
+{
+public:
+    TerransPowerBoxGoV2();
+    public:
+        virtual ~TerransPowerBoxGoV2() = default;
+        virtual bool initProperties() override;
+        virtual bool updateProperties() override;
+        virtual void TimerHit() override;
+        virtual bool ISNewSwitch(const char *dev, const char *name, ISState *states, char *names[], int n) override;
+        virtual bool ISNewText(const char *dev, const char *name, char *texts[], char *names[], int n) override;
+             
+    protected:
+        virtual const char *getDefaultName() override;
+        virtual bool saveConfigItems(FILE *fp) override;
+        
+    private:           
+        bool Handshake();
+        bool sendCommand(const char * cmd, char * res);
+        int PortFD{-1};
+       
+        bool setupComplete { false };
+        bool processButtonSwitch(const char *dev, const char *name, ISState *states, char *names[],int n);
+        
+        void Get_State();
+
+        Connection::Serial *serialConnection { nullptr };
+        
+        const char *ConfigRenameDCA { nullptr };
+        const char *ConfigRenameDCB { nullptr };
+        const char *ConfigRenameDCC { nullptr };
+        const char *ConfigRenameDCD { nullptr };
+        const char *ConfigRenameDCE { nullptr };
+        const char *ConfigRenameUSBA { nullptr };
+        const char *ConfigRenameUSBB { nullptr };
+        const char *ConfigRenameUSBE { nullptr };
+        const char *ConfigRenameUSBF { nullptr };
+         
+        //Power Switch
+        ISwitch DCAS[2];
+        ISwitch DCBS[2];
+        ISwitch DCCS[2];
+        ISwitch DCDS[2];
+        ISwitch DCES[2];
+        
+        ISwitch USBAS[2];
+        ISwitch USBBS[2];
+        ISwitch USBES[2];
+        ISwitch USBFS[2];
+        
+        ISwitch StateSaveS[2];     
+           
+        ISwitchVectorProperty DCASP;
+        ISwitchVectorProperty DCBSP;
+        ISwitchVectorProperty DCCSP;
+        ISwitchVectorProperty DCDSP;
+        ISwitchVectorProperty DCESP;
+        
+        ISwitchVectorProperty USBASP;
+        ISwitchVectorProperty USBBSP;
+        ISwitchVectorProperty USBESP;
+        ISwitchVectorProperty USBFSP;
+        
+        ISwitchVectorProperty StateSaveSP;
+        
+        //Sensor Date
+        INumber InputVotageN[1];
+        INumberVectorProperty InputVotageNP;
+        
+        INumber InputCurrentN[1];
+        INumberVectorProperty InputCurrentNP;
```

Comment:
Please use new INDI::PropertySwitch, INDI::PropertyText, INDI::PropertyNumber ...etc as the old style properties (e.g. IXXXProperty..etc are deprecated).